### PR TITLE
Major and minor fixes

### PR DESCRIPTION
**Pull request by neilprado**

### CODE REVIEW

File: heimdall-gateway/src/main/java/br/com/conductor/heimdall/gateway/filter/helper/CallImpl.java [hunk 1]
```diff
@@ -465,8 +459,6 @@ public void setBody(byte[] body, boolean gzip) {
                  context.setResponseDataStream(stream);
                  writeResponse(stream, context.getResponse().getOutputStream());
 
-             } catch (UnsupportedEncodingException e) {
```
Comment (smrafael) [inline heimdall-gateway/src/main/java/br/com/conductor/heimdall/gateway/filter/helper/CallImpl.java:468]: Why this catch was removed?

File: heimdall-gateway/src/main/java/br/com/conductor/heimdall/gateway/filter/helper/DBMongoImpl.java [hunk 2]
```diff
@@ -294,12 +295,14 @@ private MongoDatabase database() {
...
-          } catch (IllegalArgumentException | IllegalAccessException e) {
-               log.error(e.getMessage(), e);
+          Optional<Field> idOptional = Arrays.stream(object.getClass().getDeclaredFields()).filter(field -> field.getAnnotation(Id.class) != null).findFirst();
+          Field id;
+          if(idOptional.isPresent()){
+               id = idOptional.get();
+               id.setAccessible(true);
+               try {
+                    return id.get(object);
+               } catch (IllegalArgumentException | IllegalAccessException ignored) { }
```
Comment (smrafael) [inline heimdall-gateway/src/main/java/br/com/conductor/heimdall/gateway/filter/helper/DBMongoImpl.java:305]: Why you are ignoring the exception?

File: heimdall-gateway/src/main/java/br/com/conductor/heimdall/gateway/filter/ScopesFilter.java [hunk 3]
```diff
@@ -72,7 +72,7 @@ public Object run() {
         try {
             process();
             detail.setStatus(Constants.SUCCESS);
-        } catch (Throwable e) {
+        } catch (Exception e) {
```
Comment (smrafael) [inline heimdall-gateway/src/main/java/br/com/conductor/heimdall/gateway/filter/ScopesFilter.java:75]: Revert.

File: heimdall-gateway/src/main/java/br/com/conductor/heimdall/gateway/filter/HeimdallFilter.java [hunk 4]
```diff
@@ -97,7 +99,7 @@ public Object run() {
           return null;
      }     
 
-     public abstract void execute() throws Throwable;
+     public abstract void execute() throws IOException;
```
Comment (smrafael) [inline heimdall-gateway/src/main/java/br/com/conductor/heimdall/gateway/filter/HeimdallFilter.java:102]: Verify if this change is valid.

Comment (neilprado) [reply heimdall-gateway/src/main/java/br/com/conductor/heimdall/gateway/filter/HeimdallFilter.java]: This change is not valid. Reverted

File: heimdall-gateway/src/main/java/br/com/conductor/heimdall/gateway/filter/ScopesFilter.java (hunk 3 above)
Comment (neilprado) [reply heimdall-gateway/src/main/java/br/com/conductor/heimdall/gateway/filter/ScopesFilter.java]: Reverted

File: heimdall-gateway/src/main/java/br/com/conductor/heimdall/gateway/filter/helper/DBMongoImpl.java (hunk 2 above)
Comment (neilprado) [reply heimdall-gateway/src/main/java/br/com/conductor/heimdall/gateway/filter/helper/DBMongoImpl.java]: Reverted

File: heimdall-gateway/src/main/java/br/com/conductor/heimdall/gateway/filter/helper/CallImpl.java (hunk 1 above)
Comment (neilprado) [reply heimdall-gateway/src/main/java/br/com/conductor/heimdall/gateway/filter/helper/CallImpl.java:468]: Reverted

**Pull request merged**